### PR TITLE
Fix serialization

### PR DESCRIPTION
**Pull request by yoikawa**
- moved `Noun#makeSkinny()` to super class (`GrammaticalTerm`) so that `Adjective` and `Article` also have a chance to make itself 'skinny'
- replaced remaining `String#intern()` with `IniFileUtil#intern(String)`

note: pls use "Hide whitespace" option

### CODE REVIEW

File: src/main/java/com/force/i18n/grammar/AbstractLanguageDeclension.java [hunk 1]
```diff
@@ -155,13 +155,13 @@ public EnumSet<LanguageCase> getAllowedCases() {
...
 
-    final static EnumSet<LanguageStartsWith> CONSONANT_SET = EnumSet.of(LanguageStartsWith.CONSONANT);
+    static final EnumSet<LanguageStartsWith> CONSONANT_SET = EnumSet.of(LanguageStartsWith.CONSONANT);
     @Override
     public EnumSet<LanguageStartsWith> getRequiredStartsWith() {
         return CONSONANT_SET;  // Only generally care about consonant.
     }
 
-    final static EnumSet<LanguagePossessive> POSSESSIVE_NONE_SET = EnumSet.of(LanguagePossessive.NONE);
+    static final EnumSet<LanguagePossessive> POSSESSIVE_NONE_SET = EnumSet.of(LanguagePossessive.NONE);
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/AbstractLanguageDeclension.java:164]: reordered modifiers by following recommendation from Sonarlint ([java:S1124](https://rules.sonarsource.com/java/RSPEC-1124))

File: src/main/java/com/force/i18n/grammar/AbstractLanguageDeclension.java [hunk 2]
```diff
@@ -462,7 +461,7 @@ public final LanguagePluralRules getPluralRules() {
      * Often times, complex declension have simple forms of one thing or another.
      * This is the "simple" form types for which there is only one possible declension
      */
-    public static enum SimpleModifierForm implements AdjectiveForm, ArticleForm {
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/AbstractLanguageDeclension.java:465]: removed redundant `static` modifier for inner-enum class ([java:S2786](https://rules.sonarsource.com/java/RSPEC-2786))

File: src/main/java/com/force/i18n/grammar/AbstractLanguageDeclension.java [hunk 3]
```diff
@@ -609,10 +609,6 @@ public SimplePluralNounWithGender(LanguageDeclension declension, String name, St
             super(declension, name, pluralAlias, type, entityName, LanguageStartsWith.CONSONANT, gender, access, isStandardField, isCopiedFromDefault);
         }
 
-        @Override
-        public void makeSkinny() {
-        }
-
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/AbstractLanguageDeclension.java:615]: this has been moved to super class -- [GrammaticalTerm.java](https://github.com/salesforce/grammaticus/pull/38/files?diff=unified&w=1#diff-fdaa532f473c1449f403a7bee5c58d9666990dcde583072f8db65e5f7c84de17R131-R133)

File: src/main/java/com/force/i18n/grammar/LanguageDictionary.java [hunk 4]
```diff
@@ -64,12 +64,12 @@ public final class LanguageDictionary implements Serializable {
     // instances; but it would have to be concurrent.
     // TODO: "TableEnumOrId" should be stored on each Noun, right?
     /** For UI support. keyed by TableEnumOrId to HashMap(name, NounType) */
-    private final Multimap<String, Noun> nounsByEntityType;
+    private transient Multimap<String, Noun> nounsByEntityType;
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/LanguageDictionary.java:67]: see `[#readObject(ObjectInputStream)](https://github.com/salesforce/grammaticus/pull/38/files#diff-91aec7f03f9d533b88cf12eeac4f2771e64c2080b9a1992481bf30c17b8139a5L644)` below.

File: src/main/java/com/force/i18n/grammar/LanguageDictionary.java [hunk 5]
```diff
@@ -94,20 +94,29 @@ public LanguageDictionary(HumanLanguage language, LanguageDictionary from) {
 
     @Override
     public boolean equals(Object o) {
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/LanguageDictionary.java:96]: fixed  `equals(Object)` implementation that returned inaccurate result although it's generally not used

File: src/main/java/com/force/i18n/grammar/LanguageDictionary.java [hunk 6]
```diff
@@ -338,12 +352,12 @@ public void put(String name, GrammaticalTerm term) {
         if (term instanceof Noun) {
             Noun noun = (Noun) term;
             assert name.equals(((Noun)term).getName()) : "Trying to put a noun into the map at the wrong name";
-            nounMap.put(name.intern(), noun);
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/LanguageDictionary.java:341]: stop using `String#intern()`

File: src/main/java/com/force/i18n/grammar/NounForm.java [hunk 7]
```diff
@@ -33,11 +33,4 @@ public interface NounForm extends GrammaticalForm {
      * @return the possessive associated with this noun form
      */
     LanguagePossessive getPossessive();
-
-    /**
-     * @return a HTML compatible screen that can be used to represent this noun
-     * form uniquely when compared to all other noun forms.
-     */
-    @Override
-    String getKey();
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/NounForm.java:42]: this is redundant. the super class (interface) `GrammaticalForm` already has the same method signature

File: src/main/java/com/force/i18n/grammar/ArticledDeclension.java [hunk 8]
```diff
@@ -21,11 +21,11 @@
  * @author stamm
  */
 public abstract class ArticledDeclension extends AbstractLanguageDeclension {
-    public ArticledDeclension(HumanLanguage language) {
-		super(language);
-	}
+    protected ArticledDeclension(HumanLanguage language) {
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/ArticledDeclension.java:24]: constructor of `abstract class` should not be `public`

File: src/main/java/com/force/i18n/grammar/parser/GrammaticalLabelFileHandler.java [hunk 9]
```diff
@@ -174,10 +176,11 @@ abstract class BaseTag {
             this.parent = parent;
             this.attributes = atts;
             if (atts != null) {
-                this.name = getParser().uniquefy(atts.getValue(NAME));
+                String n = atts.getValue(NAME);
+                this.name = (n != null) ? intern(n.trim()) : null;
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/parser/GrammaticalLabelFileHandler.java:180]: trim `name` attribute value here. 
e.g. `<param name="some value ">some label</param>`

File: src/main/java/com/force/i18n/grammar/Noun.java [hunk 10]
```diff
@@ -357,48 +356,24 @@ public String toString() {
...
-    	if (getDeclension().hasClassifiers() && getClassifier() != null) {
-    	    appendable.append("\"c\":\"").append(getClassifier()).append("\",");
-    	}
-    	appendable.append("\"v\":{");
-    	Map<? extends NounForm,String> allValues = getAllDefinedValues();
-    	appendable.append(new TreeMap<>(allValues).entrySet().stream().map(e->"\""+e.getKey().getKey()+"\":\""+TextUtil.escapeForJsonString(e.getValue())+"\"").collect(Collectors.joining(",")));
-    	appendable.append("}}");
-	}
-
-	/**
```
Comment (yoikawa) [inline src/main/java/com/force/i18n/grammar/Noun.java:385]: moved this to super class

**Review by apoblock — APPROVED** [review]
lgtm

**Pull request merged**